packages/compass-sidebar/src/modules/databases.ts

```diff
@@ -45,7 +46,8 @@ export type Database = Pick<
     '_id' | 'name' | 'type' | 'sourceName' | 'pipeline'
   >[];
 };
-export interface DatabaseState {
+export type DatabaseState = Record<ConnectionInfo['id'], SingleDatabaseState>;
+export interface SingleDatabaseState {
```

Comment:
This sounds very much like there is a single Database, but it's actually a list 😄. Perhaps we can use something like `AllDatabasesState` and `ConnectionDatabasesState`?

Reply:
yeah, I totally agree, I think I'll rename it as you suggested.